### PR TITLE
DX: Allow Token class constructor arrows with exactly 2 elements

### PR DESCRIPTION
**Pull request by kubawerlos**

### CODE REVIEW

File: src/Tokenizer/Token.php [hunk 1]
```diff
@@ -56,6 +56,10 @@ class Token
     public function __construct($token)
     {
         if (\is_array($token)) {
+            if (2 !== \count($token)) {
```
Comment (keradus) [inline src/Tokenizer/Token.php:59]: :-1:

I believe it's super handy to pass 3-items array as well, eg directly from `token_get_all`, like
`new Token(token_get_all($codeAsAString)[$index]);`

mapping 3-items array to 2 items array, like you need to do later, is not only tedious, but also wasting cpu/ram in `Tokens` class

Comment (keradus) [reply src/Tokenizer/Token.php:59]: pls change to 2 or 3 elements in array

File: src/Tokenizer/Tokens.php [hunk 2]
```diff
@@ -1105,7 +1105,7 @@ public function setCode($code)
         $this->setSize(\count($tokens));
 
         foreach ($tokens as $index => $token) {
-            $this[$index] = new Token($token);
+            $this[$index] = new Token(\is_array($token) ? [$token[0], $token[1]] : $token);
```
Comment (mvorisek) [inline src/Tokenizer/Tokens.php:1108]: why is this helpful?

File: src/Tokenizer/Token.php (hunk 1 above)
Comment (kubawerlos) [reply src/Tokenizer/Token.php:59]: The idea of this PR was to restrict 2 or 3 number to allowing only 2, if we want to allow 2 or 3 the this check is not needed at all.

**Pull request closed without merging**